tests/integration/test_charm_ambient.py

```diff
@@ -0,0 +1,414 @@
+
+
+# Parameterize to two different profiles?
+async def test_profile_creation(lightkube_client, profile):
+    """Test whether a namespace was created for this profile."""
+    profile_name = profile
+    validate_profile_namespace(lightkube_client, profile_name, DEFAULT_SECURITY_POLICY)
+
+
+async def test_config_option_propagation(ops_test, lightkube_client, profile):
+    """Test that changes to the security policy are properly propagated."""
+    new_security_policy = "baseline"
+    await ops_test.model.applications[CHARM_NAME].set_config(
+        {"security-policy": new_security_policy}
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[CHARM_NAME], status="active", raise_on_blocked=True, timeout=600
+    )
+
+    profile_name = profile
+    validate_profile_namespace(lightkube_client, profile_name, new_security_policy)
+
+    # Change back to the default value
+    await ops_test.model.applications[CHARM_NAME].set_config(
+        {"security-policy": DEFAULT_SECURITY_POLICY}
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[CHARM_NAME], status="active", raise_on_blocked=True, timeout=600
+    )
+
+    profile_name = profile
+    validate_profile_namespace(lightkube_client, profile_name, DEFAULT_SECURITY_POLICY)
+
+
+@retry(wait=wait_exponential(multiplier=1, min=1, max=10), stop=stop_after_delay(60), reraise=True)
+async def test_waypoint_gateway_created_for_profile(
+    ops_test: OpsTest, lightkube_client: lightkube.Client, profile
+):
+    """Test whether the waypoint gateway is created in the profile namespace."""
+    gateway_objects = list(lightkube_client.list(res=GATEWAY, namespace=profile))
+    assert (
+        len(gateway_objects) == 1
+    ), f"Expected 1 Gateway object in namespace '{profile}', found {len(gateway_objects)}"
+
+
+async def test_authorization_policy_has_correct_principals(
+    ops_test: OpsTest, lightkube_client: lightkube.Client, profile
+):
+    """Test whether the authorization policy contains the three configured principals."""
+
+    @retry(
+        wait=wait_exponential(multiplier=1, min=1, max=10), stop=stop_after_delay(60), reraise=True
+    )
+    def get_auth_policy():
+        return lightkube_client.get(
+            res=AUTHORIZATION_POLICY, name=AUTH_POLICY_NAME, namespace=profile
+        )
+
+    ns_owner_auth_policy = get_auth_policy()
+
+    # Convert the policy to a string to search for principals
+    policy_str = str(ns_owner_auth_policy.to_dict())
+
+    # Check that all three principals are mentioned in the policy
+    assert (
+        ISTIO_GATEWAY_PRINCIPAL in policy_str
+    ), f"Expected principal {ISTIO_GATEWAY_PRINCIPAL} not found in authorization policy"
+    assert (
+        JUPYTER_CONTROLLER_PRINCIPAL in policy_str
+    ), f"Expected principal {JUPYTER_CONTROLLER_PRINCIPAL} not found in authorization policy"
+    assert (
+        KFP_UI_PRINCIPAL in policy_str
+    ), f"Expected principal {KFP_UI_PRINCIPAL} not found in authorization policy"
```

Comment:
nitpick: we could loop through the constants in order not to repeat them twice in every assertion